src/main/java/org/apache/cassandra/sidecar/CQLSession.java

```diff
@@ -73,19 +98,66 @@ public synchronized Session getLocalCql()
         {
             if (localSession == null)
             {
-                cluster = Cluster.builder()
-                                 .addContactPointsWithPorts(inet)
-                                 .withLoadBalancingPolicy(wlp)
-                                 .withQueryOptions(queryOptions)
-                                 .withReconnectionPolicy(reconnectionPolicy)
-                                 .withoutMetrics()
-                                 // tests can create a lot of these Cluster objects, to avoid creating HWTs and
-                                 // event thread pools for each we have the override
-                                 .withNettyOptions(nettyOptions)
-                                 .build();
+                Cluster.Builder builder = Cluster.builder()
+                                                 .addContactPointsWithPorts(address)
+                                                 .withLoadBalancingPolicy(wlp)
+                                                 .withQueryOptions(queryOptions)
+                                                 .withReconnectionPolicy(reconnectionPolicy)
+                                                 .withoutMetrics()
+                                                 // tests can create a lot of these Cluster objects, to avoid creating
+                                                 // HWTs and event thread pools for each we have the override
+                                                 .withNettyOptions(nettyOptions);
+
+                if (useSSL)
+                {
+                    try
+                    {
+                        if (trustStorePath != null && trustStorePassword != null)
```

Comment:
I think this might be required for future tests as well. Perhaps consider adding this into a utility class?